sql/importer: TestImportIntoCSV failed

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9371042?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9371042?buildTab=artifacts#/) on master @ [a6b9ad50967f49a2fde23b24a0c0cebb87c1c188](https://github.com/cockroachdb/cockroach/commits/a6b9ad50967f49a2fde23b24a0c0cebb87c1c188):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 1574471 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestImportIntoCSV
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1bb943832b8a00effe98912a1a7e8043/logTestImportIntoCSV1410107140
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestImportIntoCSV/import-into-auto-bzip
=== RUN   TestImportIntoCSV/import-into-explicit-gzip
=== RUN   TestImportIntoCSV/import-into-explicit-gzip-wildcard
=== RUN   TestImportIntoCSV/import-into-no-database
=== RUN   TestImportIntoCSV/import-into-no-decompress
=== RUN   TestImportIntoCSV/empty-with-files
=== RUN   TestImportIntoCSV/import-into-bad-opt-name
=== RUN   TestImportIntoCSV/import-into-explicit-bzip
=== RUN   TestImportIntoCSV/import-into-no-decompress-gzip#01
=== RUN   TestImportIntoCSV/import-into-no-decompress-wildcard
=== RUN   TestImportIntoCSV/empty-file
=== RUN   TestImportIntoCSV/import-into-implicit-bzip
=== RUN   TestImportIntoCSV/import-into-implicit-gzip
=== RUN   TestImportIntoCSV/import-into-no-table
=== RUN   TestImportIntoCSV/import-no-files-match-wildcard
=== RUN   TestImportIntoCSV/import-into-auto-gzip
=== RUN   TestImportIntoCSV/import-into-auto-bzip-wildcard
=== RUN   TestImportIntoCSV/import-into-auto-decompress
=== RUN   TestImportIntoCSV/import-into-no-decompress-gzip
=== RUN   TestImportIntoCSV/import-into-no-glob-wildcard
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26389

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9372847?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9372847?buildTab=artifacts#/) on master @ [99102ddf4b7602788b422366f1acc14b81c64d24](https://github.com/cockroachdb/cockroach/commits/99102ddf4b7602788b422366f1acc14b81c64d24):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 1108821 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestImportIntoCSV
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1bb943832b8a00effe98912a1a7e8043/logTestImportIntoCSV2735019186
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestImportIntoCSV/import-into-no-decompress-gzip#01
=== RUN   TestImportIntoCSV/import-into-no-glob-wildcard
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9377179?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9377179?buildTab=artifacts#/) on master @ [99102ddf4b7602788b422366f1acc14b81c64d24](https://github.com/cockroachdb/cockroach/commits/99102ddf4b7602788b422366f1acc14b81c64d24):


```
=== RUN   TestImportIntoCSV
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestImportIntoCSV1103499802
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestImportIntoCSV
    import_stmt_test.go:3110: error executing 'RESUME JOB $1': pq: job 852880421668225025: job with status pause-requested cannot be resumed
=== CONT  TestImportIntoCSV
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestImportIntoCSV1103499802
--- FAIL: TestImportIntoCSV (200.19s)
=== RUN   TestImportIntoCSV/import-into-no-glob-wildcard
=== CONT  TestImportIntoCSV/import-into-no-glob-wildcard
    testing.go:1343: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
    --- FAIL: TestImportIntoCSV/import-into-no-glob-wildcard (5.84s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/9389435?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests_BazelUnitTests/9389435?buildTab=artifacts#/) on master @ [1e27c65fb5a3810a5e2285cf07b45d95d401cca6](https://github.com/cockroachdb/cockroach/commits/1e27c65fb5a3810a5e2285cf07b45d95d401cca6):


```
=== RUN   TestImportIntoCSV
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/bbf4d7cc679fe1381d7566bc672ef5f5/logTestImportIntoCSV739500722
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestImportIntoCSV
    import_stmt_test.go:3110: error executing 'RESUME JOB $1': pq: job 853101888310902785: job with status pause-requested cannot be resumed
=== CONT  TestImportIntoCSV
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/bbf4d7cc679fe1381d7566bc672ef5f5/logTestImportIntoCSV739500722
--- FAIL: TestImportIntoCSV (12.46s)
=== RUN   TestImportIntoCSV/import-into-no-glob-wildcard
=== CONT  TestImportIntoCSV/import-into-no-glob-wildcard
    testing.go:1343: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
    --- FAIL: TestImportIntoCSV/import-into-no-glob-wildcard (1.71s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9386985?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9386985?buildTab=artifacts#/) on master @ [1e27c65fb5a3810a5e2285cf07b45d95d401cca6](https://github.com/cockroachdb/cockroach/commits/1e27c65fb5a3810a5e2285cf07b45d95d401cca6):


```
=== RUN   TestImportIntoCSV
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestImportIntoCSV4059844223
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestImportIntoCSV
    import_stmt_test.go:3110: error executing 'RESUME JOB $1': pq: job 853163646067179521: job with status pause-requested cannot be resumed
=== CONT  TestImportIntoCSV
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestImportIntoCSV4059844223
--- FAIL: TestImportIntoCSV (21.92s)
=== RUN   TestImportIntoCSV/import-into-no-glob-wildcard
=== CONT  TestImportIntoCSV/import-into-no-glob-wildcard
    testing.go:1343: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
    --- FAIL: TestImportIntoCSV/import-into-no-glob-wildcard (2.06s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestImportIntoCSV [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9402827?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9402827?buildTab=artifacts#/) on master @ [82284bdc510a12c12e65fa10504f12fa2ec33529](https://github.com/cockroachdb/cockroach/commits/82284bdc510a12c12e65fa10504f12fa2ec33529):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 1358640 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x88
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestImportIntoCSV
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1bb943832b8a00effe98912a1a7e8043/logTestImportIntoCSV9732914
    test_log_scope.go:79: use -show-logs to present logs inline
=== RUN   TestImportIntoCSV/import-into-explicit-gzip
=== RUN   TestImportIntoCSV/import-into-explicit-gzip-wildcard
=== RUN   TestImportIntoCSV/import-into-implicit-gzip
=== RUN   TestImportIntoCSV/import-into-no-decompress
=== RUN   TestImportIntoCSV/import-into-no-glob-wildcard
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportIntoCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -3024,7 +3024,6 @@ func TestImportIntoCSV(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 100477, "programming error in dropTableAfterJobComplete below")
 	skip.UnderShort(t)
 	skip.UnderRace(t, "takes >1min under race")
 
@@ -3102,15 +3101,14 @@ func TestImportIntoCSV(t *testing.T) {
 	testNum := -1
 	insertedRows := numFiles * rowsPerFile
 
-	// Some of the tests result in a failing import. in this case,
-	// the table won't be able to drop until IMPORT's
-	// OnFailOrCancel brings the table back online.
+	// Some of the tests result in a failing import. In this case, the table
+	// won't be able to drop until IMPORT's OnFailOrCancel brings the table back
+	// online.
 	//
-	// Depening on which node that the import started on,
-	// we may have paused rather than failed. This is
-	// because _all_ errors that "rpc error" are
-	// retriable. If we end up with a cross-node nodelocal
-	// request, we get a pause.
+	// Depending on which node that the import started on, we may have paused
+	// rather than failed. This is because _all_ errors that "rpc error" are
+	// retriable. If we end up with a cross-node nodelocal request, we get a
+	// pause.
 	dropTableAfterJobComplete := func(t *testing.T, tableName string) {
 		var jobID int
 		row := conn.QueryRow("SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT' AND status IN ('paused', 'pause-requested', 'reverting')")
@@ -3119,7 +3117,19 @@ func TestImportIntoCSV(t *testing.T) {
 			t.Fatal(err)
 		}
 		if jobID != 0 {
-			sqlDB.Exec(t, "RESUME JOB $1", jobID)
+			// If the job has the "pause-requested" status, we must block until
+			// it transitions to the "paused" status (because we cannot cancel
+			// the job otherwise).
+			testutils.SucceedsSoon(t, func() error {
+				r := sqlDB.QueryRow(t, "SELECT status FROM [SHOW JOBS] WHERE job_id = $1", jobID)
+				var status string
+				r.Scan(&status)
+				if status == string(jobs.StatusPauseRequested) {
+					return errors.New("still has pause-requested status")
+				}
+				return nil
+			})
+			sqlDB.Exec(t, "CANCEL JOB $1", jobID)
 			sqlDB.Exec(t, "SHOW JOB WHEN COMPLETE $1", jobID)
 		}
 		sqlDB.Exec(t, fmt.Sprintf("DROP TABLE %s", tableName))

